pkgs/applications/misc/collision/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib
+, fetchFromGitHub
+, crystal
+, libadwaita
+, gtk3
+, gobject-introspection
+, openssl
+, libxml2
+, pkg-config
+, wrapGAppsHook4
+}:
+let
+  gi_crystal = crystal.buildCrystalPackage {
+    name = "gi-crystal";
+
+    src = fetchFromGitHub {
+      owner = "hugopl";
+      repo = "gi-crystal";
+      rev = "refs/tags/v0.16.0";
+      hash = "sha256-ij4U8BoSNHpkh5SmvoH5anGB/ZdvmC5zDwJCusR/s/c=";
+    };
+
+    format = "make";
+    shardsFile = ./gi-crystal/gi-crystal-shards.nix;
+    lockFile = ./gi-crystal/gi-crystal-shard.lock;
+    patches = [./gi-crystal/gi-crystal.patch];
+
+    nativeBuildInputs = [ gobject-introspection ];
+    buildInputs = [ libadwaita ];
+
+    doCheck = false;
+    doInstallCheck = false;
```

Comment:
Do we really need to disable both?

Reply:
Without `doCheck = false;`:
```
> info - Using binding config at /build/source/src/bindings/g_object/binding.yml
       > info - Using binding config at /build/source/spec/libtest_binding.yml
       > fatal - Typelib file for namespace 'Pango', version '1.0' not found
       > make: *** [Makefile:13: test-binding] Error 1
 ```

This happens because `make test-binding` set `GI_TYPELIB_PATH="./spec/build"` for binding generator
```
test-binding: libtest generator
	GI_TYPELIB_PATH="./spec/build" LIBRARY_PATH="./spec/build" LD_LIBRARY_PATH="./spec/build" ./bin/gi-crystal spec/libtest_binding.yml -o src/auto --doc
```
Without `doInstallCheck = false;`:
```
> In /nix/store/3ba2q00y5jpjxbwcg6m9fnr3ixj3y2kx-gi-crystal/bin/compare-api:90:14
       >
       >  90 | if Process.fork
       >                  ^---
       > Warning: Deprecated Process.fork. Fork is no longer supported.
       >
       > A total of 1 warnings were found.
       > /nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 1597: [: /nix/store/3ba2q00y5jpjxbwcg6m9fnr3ixj3y2kx-gi-crystal/bin/format: unary operator expected
       > /nix/store/3ba2q00y5jpjxbwcg6m9fnr3ixj3y2kx-gi-crystal/bin/format: line 4: clang-format: not found
 ```
 I don't know what is it and just disable doInstallCheck....

Reply:
> This happens because `make test-binding` set `GI_TYPELIB_PATH="./spec/build"` for binding generator

Just remove this and see if it works.